lib/compile/codegen/code.ts

```diff
@@ -155,6 +155,19 @@ export function getProperty(key: Code | string | number): Code {
   return typeof key == "string" && IDENTIFIER.test(key) ? new _Code(`.${key}`) : _`[${key}]`
 }
 
+//Does best effort to format the name properly
+export function getEsmExportName(key: Code | string | number): Code {
```

Comment:
To be honest, I think it’s better to simply fail if user attempts to use a non-identifier as schema key, then try to adjust based on some rules that a user does not know producing unexpected names.

Reply:
Makes sense, just though I do it this way as I saw ES Build do it like this. Will change it to throw an error. Any idea for a good error message? I saw in the tests that it is possible to alias $ids 

```js
     const moduleCode = standaloneCode(ajv, {
          validateNumber: "https://example.com/number.json",
          validateString: "https://example.com/string.json",
        })
```

Maybe mention the solution like that in the error?

Reply:
Something like: invalid export name ${name}. You can use an explicit name mapping.

Do the docs not have this name mapping, is it only in tests?

Reply:
I was about to say no it doesn't, but after about 15minutes of reading the docs I see that it is mentioned only in code comments here: https://ajv.js.org/standalone.html#usage-from-code. When I initially read it last week when I started to really dig into AJV, I just didn't notice it and did not understand why it is needed so thought it isn't important. 

It might be good to somehow highlight that if `code.esm` is true then your ids must be valid/exportable names and that you can remap them like that? Otherwise you will get that error.

Reply:
yes, worth adding to the docs, but can be a separate PR